shapes/Circle.hpp

```diff
@@ -2,19 +2,21 @@
 
 #include "Shape.hpp"
 
-class Circle : public Shape
-{
+class alignas(128) Circle final : public Shape {
 public:
-    Circle(double r);
-    Circle(const Circle & other);
+    using Shape::Shape;
+    Circle(double r, Color color = Color::Green);
```

Comment:
I do not see delegating constructors here. The idea was to leave the old constructor and call the new one in the olds' initialization list:

```suggestion
    Circle(double r) : Circle(r, Color::Green) {}
    Circle(double r, Color color);
```

calling constructor from it's own class was not allowed before C++14.
The same for other classes.

PS. I know that your version is more intuitive and does the same, but idea here is to learn how does delegating constructor works.